src/payment/payments.ts

```diff
@@ -125,13 +124,14 @@ export class Payments extends EventTarget {
               provider: debitNote.provider,
             }),
           );
-          this.logger?.debug(
-            `New Debit Note received for agreement ${debitNote.agreementId}. Amount: ${debitNote.totalAmountDue}`,
-          );
+          this.logger.debug("New Debit Note received", {
+            agreementId: debitNote.agreementId,
+            amount: debitNote.totalAmountDue,
+          });
         }
       } catch (error) {
         const reason = error.response?.data?.message || error;
-        this.logger?.debug(`Unable to get debit notes. ${reason}`);
+        this.logger.error(`Unable to get debit notes.`, { reason });
```

Comment:
I'm not sure if that shouldn't be a warning. I'll see the UX when testing it

Reply:
OK, so I've checked and the result is as follows:

```
[09:52:00.585] ERROR (12208): Unable to get invoices.
    reason: "socket hang up"
```

But we can actually leave it on the error level and fix the original issue which is the broken logic of gracefully shutting down the services in the SDK, that lead to the abrupt socket hang up.